### ISSUE TITLE
Testing on SRL task

### ISSUE THREAD

**Issue by bhavitvyamalik** [nyu-mll/jiant#1250]
Hi, 
I wanted to test the data for edge probing on semantic role labelling task. I wanted something similar to validation which happens at the end of training for best saved model. For that will I have to add something like (code snippet below) for test as well? When I tried that it throws error saying it didn't find "metric" in `test_results_dict`. How can I test with my data like validation instead of writing test predictions to a file? 
```
jiant_evaluate.write_val_results(
      val_results_dict=val_results_dict,
      metrics_aggregator=runner.jiant_task_container.metrics_aggregator,
      output_dir=args.output_dir,
      verbose=True,
      name="val",
  )
```

**Comment by jeswan** [comment]
For test sets, we currently do not support evaluation metrics similar to validation sets see: https://github.com/nyu-mll/jiant/blob/605d794d89be0242f77c653677f8a84d70df7cee/jiant/proj/main/runner.py#L345 compared to https://github.com/nyu-mll/jiant/blob/605d794d89be0242f77c653677f8a84d70df7cee/jiant/proj/main/runner.py#L302. This seems like a useful feature that I'll start on. @zphang let me know if you have any concerns about adding this in.

**Comment by jeswan** [comment]
Looking into this further, it looks like the assumption that the test set does not contain labels is strongly enforced in our code base since the test predictions are generally submitted to an external script (such as a benchmark). I suggest replacing the validation set with your desired test set to obtain a metric. Let me know if you run into issues with this method of evaluation.

**Comment by bhavitvyamalik** [comment]
Yeah sure, I'll do that. Thanks a lot!
